File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int[] dp = new int[nums.length];
        dp[0] = nums[0];
        int result=dp[0];
        // dp[1] = Math.max(nums[1],dp[0]+nums[1]);
        for(int i =1;i<nums.length ;i++){
            dp[i] = Math.max(nums[i],dp[i-1]+nums[i]);
            result = Math.max(result,dp[i]);
            // temp[i] = dp[i];
            
        }
            System.out.print(result);
    return result;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxi = float('-inf')
        sumy = 0
        for i in range(len(nums)):
            sumy += nums[i]
            if sumy > maxi:
                maxi = sumy
            if sumy < 0:
                sumy = 0
        return maxi
